compiler/plugins/target/ROCM/ROCMTarget.cpp

```diff
@@ -355,35 +355,27 @@ class ROCMTargetBackend final : public TargetBackend {
     auto exportOps = llvm::to_vector_of<IREE::HAL::ExecutableExportOp>(
         variantOp.getExportOps());
     llvm::StringMap<IREE::HAL::ExecutableExportOp> exportOpMap;
-    std::vector<std::array<int32_t, 3>> workgroupSizes;
-    SmallVector<uint32_t> workgroupLocalMemories;
-    uint32_t subgroupSize = 64;
+    std::optional<uint32_t> subgroupSize;
     for (IREE::HAL::ExecutableExportOp exportOp : exportOps) {
       exportOpMap[exportOp.getSymName()] = exportOp;
 
-      std::array<int32_t, 3> workgroupSize = {1, 1, 1};
-      if (std::optional<ArrayAttr> workgroupSizeAttr =
-              exportOp.getWorkgroupSize()) {
-        for (auto [value, sizeAttr] :
-             llvm::zip_equal(workgroupSize, *workgroupSizeAttr))
-          value = cast<IntegerAttr>(sizeAttr).getInt();
-      }
-      workgroupSizes.push_back(workgroupSize);
-
+      // TODO: put this either on the variant or propagate as a function
+      // attribute instead - today this *must* be consistent across all exports
+      // and it shouldn't need to be.
       if (auto setSubgroupSize = exportOp.getSubgroupSizeAsUInt()) {
         if (setSubgroupSize.value() != 32 && setSubgroupSize.value() != 64) {
           return variantOp.emitError()
                  << "invalid subgroup size " << setSubgroupSize.value();
         }
+        if (subgroupSize.has_value() &&
+            setSubgroupSize.value() != subgroupSize.value()) {
+          return variantOp.emitError()
+                 << "multiple exports with different subgroup sizes; this is a "
+                    "limitation of the IREE compilation process and should be "
+                    "fixed";
```

Comment:
can you also print the subgroup sizes here? this should save one recompile when it fails

Reply:
👍

Reply:
oh, there's only two options (32 or 64) and so if this error is emitted it means one was 32 and one was 64